src/party_events.ts

```diff
@@ -76,3 +76,58 @@ export const getAllPartyEvents = async (fromBlock: number) => {
 
   return events;
 };
+
+export const getHaveSetNewPartyWithContributionCacheKey = (
+  partyAddress: string
+): string => `new_party_with_contribution_${partyAddress}`;
+
+export const haveSetNewPartyWithContribution = async (
+  partyAddress: string
+): Promise<boolean> => {
+  const key = getHaveSetNewPartyWithContributionCacheKey(partyAddress);
+  const res = await getRedisAsync(key);
+  if (res) {
+    console.info(
+      `already alerted about new party with contribution ${partyAddress}`
+    );
+    return true;
+  } else {
+    console.info(
+      `have NOT alerted about new party with contribution ${partyAddress}`
+    );
+    return false;
+  }
+};
+
+export const setNewPartyWithContribution = async (partyAddress: string) => {
+  const key = getHaveSetNewPartyWithContributionCacheKey(partyAddress);
+  console.info(`setting new party with contribution ${partyAddress}`);
+  await setRedisAsync(key, "true");
+};
+
+export const getHaveSetPartyHalfwayCacheKey = (partyAddress: string): string =>
```

Comment:
minor but may be good to introduce a file just for these fns that generate cache key names? not a huge deal tho

Reply:
added `redis_key_util.ts`